pykwasm/src/pykwasm/kdist/wasm-semantics/wasm.md

```diff
@@ -493,14 +512,33 @@ Upon reaching it, the label itself is executed.
 Note that, unlike in the WebAssembly specification document, we do not need the special "context" operator here because the value and instruction stacks are separate.
 
 ```k
-    syntax Instr ::= #br( Int ) [symbol(aBr)]
+    syntax Instr ::= #br( Int ) [symbol(aBr)] | "#br_aux" "(" Int ")" [symbol(aBr_aux)]
  // -------------------------------------------------
     rule <instrs> #br(_IDX) ~> (_S:Stmt => .K) ... </instrs>
```

Comment:
This also pop things until we find something different than a statment.

Reply:
look for other labels like in `if` and `loop`

Reply:
1) Find correctly the label you want to target. (Check the pop mechanism) -> br, if, loop.
2) Save all the K cell when inserting a label on the stack.
3) When finding a branch instruction, delete the whole K cell, get the correct label to branch to, and populate the K cell with the correct content without having to pop statements from `<instrs>`.  This is a potential place to pop out the label from the stack as well. Remember to pop recursively. Think about List.